activity: Add view for a remote server's audit logs.

laurynmm opened the pull request:
Adds an activity view for a remote server's audit logs.

**Notes**:
- Links to the audit log view via the remote support view in the section with the remote server user counts.
- Activity charts can be sorted by the headers, so one can sort by the "Event time" header to see a chronological view of the audit logs.
- Because this view shows both remote server and remote realm audit logs, the "Audit log ID" column can have duplicates.
- If the "Remote realm host" or "Role count: human" fields are blank, then they aren't applicable (e.g. the audit log is for the remote server or the event type isn't a synced billing events). When that data isn't present (e.g. audit logs created for bulk user creation or pre-8.0 remote servers without remote realm objects), then a string with "N/A" is shown to indicate that missing data.

---

**Screenshots and screen captures:**

<details>
<summary>Remote server support view</summary>

![Screenshot from 2024-09-16 15-30-10](https://github.com/user-attachments/assets/25b75834-c685-4c03-ba40-0edce4f38a2f)
</details>
<details>
<summary>Example of remote server audit log view from dev environment</summary>

![Screenshot from 2024-09-16 15-58-58](https://github.com/user-attachments/assets/fbdd3297-1978-472f-b7ad-6fd39b743af0)
</details>

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented:
Probably the `role_count` display could skip the values that are 0 just for cleaner display.

timabbott commented:
For the `AUDIT_LOG_ID` column, probably doing a prefix so it is, say, `S{id}` for the rare server-level ones would be a good practice to avoid duplicates in a sortable key.

laurynmm commented:
**Updates**:
- Added an "S" prefix to the audit log IDs for remote server audit logs.
- Updated role count data to only show values that are greater than 0.
- Added a key to the top header section for user roles as a quick reference for admins.

<details>
<summary>Updated screenshot of audit log view</summary>

![Screenshot from 2024-09-17 17-25-24](https://github.com/user-attachments/assets/feaeaf8e-82b2-4390-b55e-ee541cc8edad)
</details>

laurynmm commented on corporate/views/audit_logs.py:
```diff
@@ -0,0 +1,102 @@
+from zerver.models.realm_audit_logs import AbstractRealmAuditLog, AuditLogEventType
+from zilencer.models import RemoteRealmAuditLog, RemoteZulipServer, RemoteZulipServerAuditLog
+
+USER_ROLES_KEY = "100: owner, 200: admin, 300: moderator, 400: member, 600: guest"
```
This will likely be useful in other audit log views that we implement for realms/users.

timabbott commented:
Merged, thanks @laurynmm!
